code/processes/dqe.q

```diff
@@ -56,6 +56,14 @@ nullchk:{[t;colslist;thres]
   update thresholdfail:nullspercentage>thres from res                                                           /- compare each column's nulls percentage with threshold thres
   }
 
+rangechk:{t;colslist;tlower;tupper;thres]                                                                       /- function to check percentage of values within tlower and tupper range
+  colslist:colslist except exec c from meta t where t in "csSC";                                                /- exclude columns that do not have pre-defined limits
+  tupper:((),colslist)#tupper;
+  tlower:((),colslist)#tlower;
+  d:(sum (tt>=tlower)&(tt<=tupper))*100%count tt:((),colslist)#t;                                               /- dictionary with results by columns
```

Comment:
can you use `within` here instead?

Also I may be misinterpreting the data structures here, but does this actually work?

```q
q)colslist:`a`b
q)t:([] a:til 30;b:til 30;d:til 30)
q)tt:colslist#t
q)tlower:`a`b`c!10 15 20
q)tlower:colslist#tlower
q)tt>=tlower
'length
  [0]  tt>=tlower
         ^
q)
```

Reply:
Thanks, tlower and tupper are tables not dictionaries, so for your example it can be
 q)tlower:([] a:30#10;b:30#15;c:30#20). Then after selecting columns in colslist, we have tlower:colslist#tlower and then q)tt>=tlower returns table we are looking for: 
q)tt>=tlower
a b
---
0 0
0 0
0 0
0 0
0 0
0 0
0 0
0 0
0 0
0 0
1 0
1 0
1 0
1 0
1 0
1 1
1 1
1 1
1 1
1 1
..

Would it be better to use another data type for tlower/tupper?

Reply:
a table seems a bit non-obvious to me, I'd have expected a dictionary given there's presumably only going to be one lower limit value per column. at the very least, this needs documented somewhere etc.

Reply:
My reasoning was to have functionality of possible multiple upper and lower bounds for each column (such as different limits for different timepoints).
I've added comments about tlower and tupper being tables but can change to a dictionary as a better option if we assume one limit per column?